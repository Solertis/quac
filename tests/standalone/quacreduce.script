#!/bin/bash

# Test that the quacreduce script works, using shell commands for map and
# reduce (see quacreduce-python test for the Python version). Section numbers
# refer to the "Map-Reduce with quacreduce" section in the documentation; make
# sure these two files match.
#
# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

. ./environment.sh

cd $DATADIR


## Test map-reduce using shell pipeline

echo '4.2.1. Create sample input'
echo -e 'foo bar baz\nfoo foo' > foo1.txt
echo -e 'bar' > foo2.txt
cat foo*.txt

echo '4.2.2. Define the map operator'
cat foo*.txt | tr '[:blank:]' '\n'

echo '4.2.3. Define the reduce operator'
echo -e 'b\nb\na\nc\nc\nc' | uniq -c

echo '4.2.4. Test the operators together'
cat foo*.txt | tr '[:blank:]' '\n' | sort -sk1 -t '	' | uniq -c


## Test map-reduce using quacreduce

echo '4.2.5 Prepare the job'
quacreduce --map 'tr "[:blank:]" "\n"' \
           --reduce 'uniq -c > out/%(RID)' \
           --partitions 2 \
           foo*.txt

echo '4.2.6 Run the job with make'
x ls -R .
x make --quiet  # output contains temp dirs that vary
x ls -R .
y "cat out/* | sort"

#echo '4.2.8 Add more input data'
#echo 'qux' > foo3.txt
#quacreduce --update job /tmp/foo*.txt
#(cd job && make -j2)
#cat out/*


## Test cleanup

echo '*** Test cleanup'
y 'make --quiet clean'
x ls -R .
y 'make --quiet reallyclean'
x ls -R .
