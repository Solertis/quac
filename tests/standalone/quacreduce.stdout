4.2.1. Create sample input
foo bar baz
foo foo
bar
4.2.2. Define the map operator
foo
bar
baz
foo
foo
bar
4.2.3. Define the reduce operator
      2 b
      1 a
      3 c
4.2.4. Test the operators together
      2 bar
      1 baz
      3 foo
4.2.5 Prepare the job
4.2.6 Run the job with make
$ ls -R .
.:
foo1.txt
foo2.txt
Makefile
out
tmp

./out:

./tmp:
$ make --quiet
$ ls -R .
.:
foo1.txt
foo2.txt
Makefile
out
tmp

./out:
0
1

./tmp:
0.reduced
1.reduced
foo1.txt.0
foo1.txt.1
foo1.txt.mapped
foo2.txt.0
foo2.txt.1
foo2.txt.mapped
$ (cat out/* | sort)
      1 baz
      2 bar
      3 foo
*** Test cleanup
$ (make --quiet clean)
$ ls -R .
.:
foo1.txt
foo2.txt
Makefile
out
tmp

./out:
0
1

./tmp:
$ (make --quiet reallyclean)
$ ls -R .
.:
foo1.txt
foo2.txt
Makefile
out
tmp

./out:

./tmp:
